### PR TITLE
[#124] Include TrailingNewlines check

### LINKED ISSUE #124: Include TrailingNewlines check

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #124

### CODE REVIEW

**Comment by chshersh** [comment]
Tests are passing, but I don't know why `hintman` is not triggered by this PR :thinking: :disappointed:

**Comment by chshersh** [comment]
Looks like it doesn't work correctly... See this PR:

* https://github.com/kowainik/hintman-target/pull/27

Need to do more tests...

**Comment by chshersh** [comment]
Okay, it's a separate issue, the problem is not in newline trailing handling.

File: src/Hintman/Core/Hint.hs [hunk 1]
```diff
@@ -30,24 +27,11 @@ parseHintType :: Text -> Maybe HintType
...
-    , suggestionLine   :: Line
-    , suggestionChange :: LineChange
-    } deriving (Eq, Show)
-
-data LineChange = LineChange
-    { lineChangeNew     :: Text
-    , lineChangeComment :: Maybe Text
-    , lineChangeType    :: HintType
-    } deriving (Eq, Show)
-
```
Comment (vrom911) [inline src/Hintman/Core/Hint.hs]: Thanks for the cleanup! I see that this is not needed anymore and we can reuse existing types.

File: src/Hintman/Hint.hs [hunk 2]
```diff
@@ -39,6 +48,8 @@ getModifiedFiles prInfo@PrInfo{..} = traverse toModifiedFile $ filter ((/=) Dele
     toModifiedFile :: FileDelta -> m ModifiedFile
     toModifiedFile mfDelta@FileDelta{..} = do
         let mfPath = toString fileDeltaDestFile
+        -- TODO: store 'ByteString' here instead of 'Maybe ByteString'
+        -- if we can't retrieve file content, can as well just ignore it
```
Comment (vrom911) [inline src/Hintman/Hint.hs]: Yeah, this is the improvement, as I check and log it in every type of hints...

File: src/Hintman/Hint/TrailingNewlines.hs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    mapMaybe createComment . getTrailingSpaceLines . toLines . decodeUtf8
+    <$> mfContent
+  where
+    -- TODO: remove duplication with 'getTrailingSpacesComments'
+    -- | Create a 'Comment' from all necessary data.
+    createComment :: Line -> Maybe Comment
+    createComment Line{..} = do
+        let commentPath = fileNameText
+        commentPosition <- getTargetCommentPosition mfDelta lineNumber
+        let commentBody = unlines [ "```suggestion", "```" ]
```
Comment (vrom911) [inline src/Hintman/Hint/TrailingNewlines.hs]: We should refactor this later to some function as this is the reusable part. Can add this to the tech debt.

**Pull request merged**